src/legacy/core_plugins/data/public/index_patterns/components/index_pattern_select.tsx

```diff
@@ -49,14 +54,22 @@ const getIndexPatterns = async (search: string, fields: string[]) => {
   return resp.savedObjects;
 };
 
-const getIndexPatternTitle = async (indexPatternId: string) => {
-  const savedObject = await chrome.getSavedObjectsClient().get('index-pattern', indexPatternId);
+const getIndexPatternTitle = async (
+  client: SavedObjectsClientContract,
+  indexPatternId: string
+): Promise<SimpleSavedObject<any>> => {
+  const savedObject = (await client.get('index-pattern', indexPatternId)) as SimpleSavedObject<any>;
   if (savedObject.error) {
     throw new Error(`Unable to get index-pattern title: ${savedObject.error.message}`);
   }
   return savedObject.attributes.title;
 };
 
+// Takes in stateful runtime dependencies and pre-wires them to the component
+export function createIndexPatternSelect(savedObjectsClient: SavedObjectsClientContract) {
```

Comment:
I see you added the stateful initializer! So cool. :-)
I guess we're going to have one in many components.
Do you have an idea for a common name? (not necessarily for this PR!)

Reply:
> Do you have an idea for a common name? (not necessarily for this PR!)

Yeah I think that's a good question. One possibility (which is the approach I took here for the time being) would be the use the same name in both places. If it's in the lifecycle, you can assume it is "pre-wired", and if it is exported statically, assume it isn't.

There are a number of reasons this isn't ideal, however (could lead to confusion when suddenly a component with the same name has a different contract when imported statically). I don't have any great ideas on that right now, but worth thinking about.

I'm more interested in how many people will prefer to have a static import which requires more configuration, vs a stateful one which doesn't. I think that remains to be seen still.